tools/config-generator/main.go

```diff
@@ -551,7 +511,7 @@ func setResourcesReqForJob(res yaml.MapSlice, data *baseProwJobTemplateData) {
 // parseBasicJobConfigOverrides updates the given baseProwJobTemplateData with any base option present in the given config.
 func parseBasicJobConfigOverrides(data *baseProwJobTemplateData, config yaml.MapSlice) {
 	(*data).ExtraRefs = append((*data).ExtraRefs, "  base_ref: "+(*data).RepoBranch)
-	var needDotdev, needGo114 bool
+	var needDotdev bool
 	for i, item := range config {
 		switch item.Key {
```

Comment:
I would suggest including a `default` case in this switch which logs the unexpected config, in case any client is still providing the deprecated "go*" values.

Reply:
This tool is actually only used by us, and I have removed all the `go114` field from our meta config files, so it's safe to deprecate it here.

Reply:
It's definitely not a requirement, so I'm ok with it. This was mostly a suggestion not only for this case but to cover all unexpected cases and root cause the issue quickly. If you feel confident with the current state, merge it.